index.js

```diff
@@ -199,7 +199,11 @@ function hash(src, filePath) {
 }
 
 function symlinkOrCopyFromCache(entry, dest, relativePath) {
-  mkdirp.sync(path.dirname(entry.outputFile));
-
-  symlinkOrCopySync(entry.cacheFile, dest + '/' + relativePath);
+  try {
```

Comment:
Looks good for cases where most directories in `paths` have 2 or more files.

Does it make sense to have `broccoli-filter` remember what directories it has and hasn't `mkdirp`d to avoid an exception here the first time we encounter the dir? (and also avoid the exdtra `symlinkOrCopySync`?